apps/qubit/modules/informationobject/actions/renameAction.class.php

```diff
@@ -22,39 +22,28 @@ class InformationObjectRenameAction extends sfAction
   // Allow modification of title, slug, and digital object filename
   public function execute($request)
   {
-    // Return 401 if unauthorized
-    if (!sfContext::getInstance()->user->isAuthenticated()
-      || !QubitAcl::check($this->resource, 'update'))
-    {
-      $this->response->setStatusCode(401);
-      return sfView::NONE;
-    }
-
     // Return 400 if incorrect HTTP method
-    if ($this->request->getMethod() != 'POST')
+    if ($request->isMethod('post'))
     {
-      $this->response->setStatusCode(400);
-      return sfView::NONE;
-    }
+      $resource = $this->getRoute()->resource;
 
-    // Internationalization needed for flash messages
-    ProjectConfiguration::getActive()->loadHelpers('I18N');
+      $renameComponent = new InformationObjectRenameComponent($this->context, 'informationobject', 'index');
+      $renameComponent->resource = $resource;
+      $renameComponent->execute($request);
 
-    // Handle rename form submission
-    if (null !== $request->rename)
-    {
-      $this->renameForm = new InformationObjectRenameForm;
+      $this->form = $renameComponent->form;
```

Comment:
I'd move the flash add and the redirect to the 'updateResource' method in the 'renameComponent', where the form is already binded and valid.